modules/compute/src/main/java/org/apache/ignite/internal/compute/messaging/ComputeMessaging.java

```diff
@@ -321,4 +321,52 @@ private void sendJobCancelResponse(
 
         messagingService.respond(senderConsistentId, jobCancelResponse, correlationId);
     }
+
+    /**
+     * Broadcasts job status request to all nodes in the cluster.
+     *
+     * @param jobId Job id.
+     * @return The current status of the job, or {@code null} if the job status no longer exists due to exceeding the retention time limit.
+     */
+    public CompletableFuture<JobStatus> broadcastStatusAsync(UUID jobId) {
+        CompletableFuture<JobStatus> result = new CompletableFuture<>();
+        CompletableFuture<?>[] futures = topologyService.allMembers().stream()
+                .map(node -> remoteStatusAsync(node, jobId).thenAccept(jobStatus -> {
+                    if (jobStatus != null) {
+                        result.complete(jobStatus);
+                    }
+                }))
+                .toArray(CompletableFuture[]::new);
+        allOf(futures).whenComplete((unused, throwable) -> {
+            if (Arrays.stream(futures).allMatch(CompletableFuture::isCompletedExceptionally)) {
+                result.completeExceptionally(new ComputeException(FAIL_TO_GET_JOB_STATUS_ERR, "Compute job status can't be retrieved."));
```

Comment:
I think we should populate the `cause` of the new exception with the exceptions from `futures`.